kitsune/wiki/views.py

```diff
@@ -1445,3 +1473,116 @@ def what_links_here(request, document_slug):
+            fallback_locale = locale
+            break
+    else:
+        # If any locale of Accpeted_Language Header is not in translated locale,
+        # Check if we have any fallback local for the locales of Accpeted_Languge Header and
+        # the document is translated into that fallback locale
+        # This locale mapping is mentioned in kitsune/settings.py
+        for locale in header_locales:
+            if NON_SUPPORTED_LOCALES.get(locale) in translated_locales:
+                fallback_locale = NON_SUPPORTED_LOCALES[locale]
+                break
+        else:
+            # If above does not match, then check if we have custom Wiki Locale mapping
+            # for the user requested locale.
+            # This custom Wiki Locale Mapping is mentioned in kitsune/wiki/config.py
+            if request.LANGUAGE_CODE in FALLBACK_LOCALES:
+                for locale in FALLBACK_LOCALES[request.LANGUAGE_CODE]:
+                    if locale in translated_locales:
+                        fallback_locale = locale
+                        break
+                else:
+                    # If above does not match, then we would check if any locale of
+                    # Accepted_Language Header is in the Custom Wiki Locale mappping
+                    for locale in header_locales:
+                        if locale in FALLBACK_LOCALES:
+                            # Check if the document is trnaslated into the fallback locale
+                            for locale in FALLBACK_LOCALES[locale]:
+                                if locale in translated_locales:
+                                    fallback_locale = locale
+                                    break
+                            else:
+                                continue
+                            break
+                    else:
+                        # If all fails, return None as fallback Locale
+                        fallback_locale = None
+
+            # If Custom Wiki Locale Mapping is not available for the locale user requesed, then we
+            # will check if We have any fallback locale available in Custom Wiki locale
+            # mapping for the locales of Accpeted_Launguage Header.
+            else:
+                for locale in header_locales:
+                    if locale in FALLBACK_LOCALES:
+                        # Check if the document is trnaslated into the fallback locale
+                        for locale in FALLBACK_LOCALES[locale]:
+                            if locale in translated_locales:
+                                fallback_locale = locale
+                                break
+                        else:
+                            continue
+                        break
+                else:
+                    # If all fails, return None as fallback Locale
+                    fallback_locale = None
+    return fallback_locale
+
+
+# Import from django.utils.translation.trans_real and changed as below
```

Comment:
Entire section rewrite:
"# Imported from django.utils.translation.trans_real with the following changes:
 # Removed lower() to get locale name as required
 # Added code to get only the locale name"